content/docs/guides/tips-n-tricks.md

```diff
@@ -0,0 +1,70 @@
+---
+title: "Tips, Tricks and Gotchas"
+weight: 4
+description: |
+  Tips and tricks for small places.
+  How to write efficient embedded code and avoid common mistakes.
+---
+
+## Ensure Concurrency
+
+TinyGo code runs on a single core, in a single thread (think `GOMAXPROCS=1`).
+This means we almost always have less threads than goroutines.
+Since scheduling in TinyGo is [cooperative](https://en.wikipedia.org/wiki/Cooperative_multitasking), a goroutine that never does IO or other blocking calls (f.ex. `time.Sleep()`) will lock the single available thread only for itself and never allow other goroutines to execute. In such cases `runtime.Gosched()` shall be used to force scheduler update.
```

Comment:
I'm not entirely sure this is good advice here. While this is technically correct, it may be better to first investigate whether it needs to lock up the CPU in the first place (using a mutex, channel, or something like that).
Something like this maybe?

> In such cases, you can use `runtime.Gosched()` as a workaround.